src/lib/UI/Module/ContentTree/NodeFactory.php

```diff
@@ -84,6 +95,7 @@ public function __construct(
         $this->repository = $repository;
         $this->siteaccessResolver = $siteaccessResolver;
         $this->maxLocationIdsInSingleAggregation = $maxLocationIdsInSingleAggregation;
+        $this->logger = $logger ?? new NullLogger();
```

Comment:
Logger is added, but I don't see it used anywhere. It should be removed, unless you forgot to use it somewhere.